test/commons/dom/is-in-tab-order.js

```diff
@@ -0,0 +1,90 @@
+    var target = queryFixture('<div id="target"></div>');
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return false for presentation element with tabindex set to non-parseable value', function () {
+    var target = queryFixture('<div id="target" tabindex="foobar"></div>');
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return false for presentation element with tabindex not set and role of natively focusable element', function () {
+    var target = queryFixture('<div id="target" role="button"></div>');
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return true for natively focusable element with tabindex 0', function () {
+    var target = queryFixture('<button id="target" tabindex="0"></button>');
+    assert.isTrue(isInTabOrder(target));
+  });
+
+  it('should return true for natively focusable element with tabindex 1', function () {
+    var target = queryFixture('<button id="target" tabindex="1"></button>');
+    assert.isTrue(isInTabOrder(target));
+  });
+
+  it('should return false for natively focusable element with tabindex -1', function () {
+    var target = queryFixture('<button id="target" tabindex="-1"></button>');
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return true for natively focusable element with tabindex not set', function () {
+    var target = queryFixture('<button id="target"></button>');
+    assert.isTrue(isInTabOrder(target));
+  });
+
+  // IE11 returns a negative tabindex for elements with tabindex set to an empty string, rather than create false positives, skip it
+  (isIE11 ? xit : it)(
+    'should return true for natively focusable element with tabindex set to empty string',
+    function () {
+      var target = queryFixture('<button id="target" tabindex=""></button>');
+      assert.isTrue(isInTabOrder(target));
+    }
+  );
+
+  it('should return true for natively focusable element with tabindex set to non-parseable value', function () {
+    var target = queryFixture(
+      '<button id="target" tabindex="foobar"></button>'
+    );
+    assert.isTrue(isInTabOrder(target));
+  });
+
+  it('should return false for disabled', function () {
+    var target = queryFixture('<button id="target" disabled></button>');
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return false for disabled natively focusable element with tabindex', function () {
+    var target = queryFixture(
+      '<button id="target" disabled tabindex="0"></button>'
+    );
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return false for hidden inputs', function () {
+    var target = queryFixture('<input type="hidden" id="target"></input>');
+    assert.isFalse(isInTabOrder(target));
+  });
+
+  it('should return false for non-element nodes', function () {
+    var target = queryFixture('<span id="target">Hello World</span>');
+    assert.isFalse(isInTabOrder(target));
+  });
+});
```

Comment:
I think one of these tests fail with the function written as is.

```suggestion

  it('should return false natively focusable hidden element', function () {
    var target = queryFixture('<button id="target" hidden></button>');
    assert.isTrue(isInTabOrder(target));
  });

  it('should return false hidden element with tabindex 1', function () {
    var target = queryFixture('<div id="target" tabindex="1" hidden></div>');
    assert.isTrue(isInTabOrder(target));
  });
});
```